include/yaramod/types/expressions.h

```diff
@@ -1209,6 +1209,50 @@ class OfExpression : public ForExpression
 	virtual TokenIt getLastTokenIt() const override { return _iterable->getLastTokenIt(); }
 };
 
+/**
+ * Class representing 'of' expression. 'of' expression is shortened version of
+ * for loop over string set with no loop body. It has hidden body which always contains just ( $ ).
+ *
+ * For example:
+ * @code
+ * all of ($str1, $str2) in (filesize-1000, filesize)
+ * @endcode
+ */
+class OfInRangeExpression : public ForExpression
```

Comment:
What I wonder about is whether we really need a completely new type of an expression or whether it wouldn't be better to have it as an optional part of `OfExpression`. Have you though about this option? What are the pros/cons compared to this?

Reply:
That is a good idea, the pros would be, that it requires less code to be written (which was already written) but also less code to be maintained. I do not really see any cons, I can try to implement it in next commit so we can decide when we see the code.